src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +696,99 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(object: any, key: string, serialzeValue: boolean) {
+    if (object != null && object.has(key) === true) {
+      let val = object.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return serialzeValue === true ? this.serializeValue(descriptorValue) : descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProps = val.properties;
+
+    // we do this so they don't get emited
+    this.serializedValues.add(val.$Prototype);
+    let $$typeof = this._getPropertyValue(objectProps, "$$typeof", false);
+    if ($$typeof !== null) {
+      this.serializedValues.add(($$typeof: any));
+      if ($$typeof.$Description != null) {
+        this.serializedValues.add(($$typeof.$Description: any));
+      }
+    }
+    let _owner = this._getPropertyValue(objectProps, "_owner", false);
+    if (_owner !== null) {
+      this.serializedValues.add((_owner: any));
+    }
+    let typeValue = this._getPropertyValue(objectProps, "type", true);
+    let keyValue = this._getPropertyValue(objectProps, "key", true);
+    let refValue = this._getPropertyValue(objectProps, "ref", true);
+    let propsValue = this._getPropertyValue(objectProps, "props", false);
+
+    if (propsValue !== null) {
+      this.serializedValues.add((propsValue: any));
+    }
+    invariant(typeValue !== null, "JSXElement type of null");
```

Comment:
You should log an error for this in the visitor. I think a user can construct a legal JavaScript program this trips this invariant.